src/makeInfinityAdder.js

```diff
@@ -5,6 +5,31 @@
  */
 function makeInfinityAdder() {
   // write code here
+  let total = 0;
+  const adder = (a) => {
+    total = a
+    if (a === undefined) {
+      return 0;
+    }
+
+    const inner = (b) => {
+      if (b === undefined) {
+        const saveTotal = total;
+        total = 0;
+
+        return saveTotal;
+      }
+      total += b;
+
+      return inner;
+    };
+
+    if (a !== undefined) {
+      return inner;
+    }
```

Comment:
There is an issue with the logic in the `adder` function. The initial call to `adder` with an argument should add that argument to the `total`. However, the current implementation on line 10 assigns the argument `a` directly to `total` which is incorrect as it overwrites the `total` instead of adding to it. Also, the return statement on line 12 should not be there; the function should return the inner function regardless of whether an argument is provided or not. The check for `undefined` should only be within the inner function to return the total sum and reset it.